modules/apps/marketplace/marketplace-settings-web/src/main/java/com/liferay/marketplace/settings/web/internal/portlet/action/DisconnectMVCResourceCommand.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.marketplace.settings.web.internal.portlet.action;
+
+import com.liferay.marketplace.settings.web.internal.configuration.MarketplaceConfigurationUtil;
+import com.liferay.marketplace.settings.web.internal.constants.MarketplaceSettingsPortletKeys;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCResourceCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCResourceCommand;
+import com.liferay.portal.kernel.security.auth.PrincipalException;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Keven Leone
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + MarketplaceSettingsPortletKeys.MARKETPLACE_SETTINGS,
+		"mvc.command.name=/marketplace_settings/disconnect"
+	},
+	service = MVCResourceCommand.class
+)
+public class DisconnectMVCResourceCommand extends BaseMVCResourceCommand {
+
+	@Override
+	protected void doServeResource(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)resourceRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		PermissionChecker permissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
```

Comment:
Since you have access to the ThemeDisplay, you can get the permission checker from it

Reply:
done